tests.py

```diff
@@ -452,6 +452,35 @@ def test_attribute_lookup(self):
         template = Template([data])
         self.assertEqual(template.name, data)
 
+    def test_template_from_json_no_error(self):
+        expected = {
+            'template': {
+                'data': [
+                    {'name': 'name', 'value': 'value'},
+                    {'name': 'name1', 'value': 'value1'},
+                ]
+            }
+        }
+        data = json.dumps(expected)
+        template = Template.from_json(data)
+        self.assertEqual(template.to_dict(), expected)
+
+    def test_template_from_json_collection(self):
+        expected = {
+            'collection': {
+                'template': {
+                    'data': [
+                        {'name': 'name', 'value': 'value'},
+                        {'name': 'name1', 'value': 'value1'},
+                    ]
+                }
+            }
+        }
+        data = json.dumps(expected)
+        with self.assertRaisesRegexp(ValueError, "Not valid Collection\+JSON template data."):
```

Comment:
instead of using assertRaisesRegexp, it's better to just assert a ValueError is raised. No real need to check the message. If you still want to assert the message you can do something like

with self.assertRaises(ValueError) as exc:
    template = Template.from_json(data)
    self.assertEqual(str(message), 'Not valid Collection+JSON template data.')